src/Ast/PhpClass.php

```diff
@@ -98,7 +109,14 @@ public function withMembers(Vector $members): PhpClass
     public function toCode(): string
     {
         $extends = is_null($this->extends) ? '' : " extends {$this->extends->toCode()}";
-        $class = $this->final ? 'final class' : 'class';
+
+        $class = 'class';
+        if($this->final) {
+            $class = 'final class';
+        } elseif( $this->abstract) {
```

Comment:
We should add a CS checker, as one would have caught this! (spaces after `if` and `elseif`, no extra space to start conditional)
```suggestion
        if ($this->final) {
            $class = 'final class';
        } elseif ($this->abstract) {
```